vmdb/spec/controllers/ops_controller/diagnostics_spec.rb

```diff
@@ -126,5 +126,17 @@
         controller.send(:log_depot_edit)
       end
     end
+
+    context "#tree_select" do
+      it "renders zone list for root node" do
+        controller.instance_variable_set(:@_response, ActionController::TestResponse.new)
+        controller.instance_variable_set(:@sb, :active_tree => :diagnostics_tree)
+        controller.instance_variable_set(:@layout, 'ops')
+        controller.instance_variable_set(:@_params, :id => 'root', :action => 'tree_select')
+        controller.send(:tree_select)
```

Comment:
@martinpovolny I'm not sure if we have any tests that are doing it right :wink: 

I've rewritten your test to do the right thing:

```diff
diff --git a/vmdb/spec/controllers/ops_controller/diagnostics_spec.rb b/vmdb/spec/controllers/ops_controller/diagnostics_spec.rb
index 2741b68..8b6acd0 100644
--- a/vmdb/spec/controllers/ops_controller/diagnostics_spec.rb
+++ b/vmdb/spec/controllers/ops_controller/diagnostics_spec.rb
@@ -58,6 +58,24 @@ end
 
 describe OpsController do
   render_views
+  context "#tree_select" do
+    it "renders zone list for root node" do
+      set_user_privileges
+      FactoryGirl.create(:vmdb_database)
+      EvmSpecHelper.create_guid_miq_server_zone
+
+      expect(MiqServer.my_guid).to be
+      expect(MiqServer.my_server).to be
+
+      session[:userid] = User.current_user.userid
+      session[:sandboxes] = { "ops" => { :active_tree => :diagnostics_tree } }
+      post :tree_select, :id => 'root', :format => :js
+
+      response.should render_template('ops/_diagnostics_zones_tab')
+      expect(response.status).to eq(200)
+    end
+  end
+
   context "::Diagnostics" do
     before do
       set_user_privileges
```

The patch above fails on master but passes on your branch.  Can you apply that rather than your test?  Also, make sure to put the test in the same place I did in my patch.  [This line in the before block](https://github.com/ManageIQ/manageiq/blob/ef98c5d8b1cbe69f421273fa25eb54cd0ee44837/vmdb/spec/controllers/ops_controller/diagnostics_spec.rb#L64) will mess up the environment you need.